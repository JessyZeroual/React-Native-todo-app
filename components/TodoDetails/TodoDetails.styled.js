import styled from "styled-components/native";
import { primaryColor, secondaryColor } from "../../constant/style-constants";

export const StyledTodoDetailsWrapper = styled.View`
  display: flex;
  align-items: center;
  height: 100%;
  background: #fff;
  border-bottom-color: ${secondaryColor};
  border-bottom-width: 1.5px;
`;
export const StyledWrapperHeader = styled.View`
  width: 100%;
  background: ${primaryColor};
`;

export const StyledHeader = styled.View`
  display: flex;
  align-items: center;
  flex-direction: row;
  width: 100%;
  background: white;
  border-top-right-radius: 40px;
  padding: 0px 20px;
`;

export const StyledTitle = styled.Text`
  padding: 15px;
  font-size: 25px;
`;
export const StyledLabel = styled.Text`
  font-size: 15px;
`;

export const StyledImageWrapper = styled.View`
  display: flex;
  align-items: center;
  justify-content: center;
  border-bottom-left-radius: 100px;

  padding: 20px;
  width: 100%;

  background: ${primaryColor};
`;

export const StyledCardWrapper = styled.View`
  width: 100%;
  padding: 0px 20px;
`;

export const StyledCardMedia = styled.View`
  width: 60px;
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 30px;
  border: 2px;
  border-color: ${secondaryColor};
`;

export const StyledCard = styled.TouchableOpacity`
  display: flex;
  align-items: center;
  flex-direction: row;
  background: #fff;
  border-radius: 10px;
  padding: 10px;
  margin: 5px 0px;
  width: 100%;
`;

export const StyledCardTitle = styled.Text`
  padding: 20px;
  font-size: 20px;
`;

export const StyledImage = styled.Image`
  border-radius: 40px;
  height: 200px;
  width: 200px;
`;

export const StyledAvatar = styled.Image`
  border-radius: 40px;
  height: 50px;
  width: 50px;
`;