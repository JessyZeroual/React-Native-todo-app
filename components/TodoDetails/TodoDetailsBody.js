import React from "react";
import Icon from "react-native-vector-icons/AntDesign";
import { users } from "../../api/users";
import { shadow } from "../../constant/style-constants";
import {
  StyledAvatar,
  StyledCardWrapper,
  StyledCardTitle,
  StyledCardMedia,
  StyledCard,
  StyledLabel,
} from "./TodoDetails.styled";

const TodoDetailsBody = ({ item, _updateTodo }) => {
  const onPressStatus = (completed) => {
    _updateTodo(item.id, item.userId, item.title, completed);
  };
  return (
    <StyledCardWrapper>
      <StyledLabel>Status:</StyledLabel>
      {item.completed ? (
        <StyledCard style={shadow} onPress={() => onPressStatus(false)}>
          <StyledCardMedia>
            <Icon name="check" color="#5cb85c" size={25}></Icon>
          </StyledCardMedia>
          <StyledCardTitle>Completed</StyledCardTitle>
        </StyledCard>
      ) : (
        <StyledCard style={shadow} onPress={() => onPressStatus(true)}>
          <StyledCardMedia>
            <Icon name="ellipsis1" color="#d9534f" size={25}></Icon>
          </StyledCardMedia>
          <StyledCardTitle>In progress</StyledCardTitle>
        </StyledCard>
      )}
      <StyledLabel>Created by:</StyledLabel>
      <StyledCard style={shadow}>
        <StyledCardMedia>
          <StyledAvatar
            source={{
              uri: users[item.userId].avatarUrl,
            }}
          />
        </StyledCardMedia>

        <StyledCardTitle>{users[item.userId].name}</StyledCardTitle>
      </StyledCard>
    </StyledCardWrapper>
  );
};

export default TodoDetailsBody;
