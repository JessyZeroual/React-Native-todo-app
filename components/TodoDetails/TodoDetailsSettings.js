import React from "react";
import * as ImagePicker from "expo-image-picker";
import Icon from "react-native-vector-icons/AntDesign";

import { primaryColor, shadow } from "../../constant/style-constants";
import {
  StyledCardMedia,
  StyledCardTitle,
  StyledCard,
} from "./TodoDetails.styled";

const TodoDetailsSettings = ({ item, addImage }) => {
  let openImagePickerAsync = async () => {
    let permissionResult = await ImagePicker.requestCameraRollPermissionsAsync();

    if (permissionResult.granted === false) {
      alert("Permission to access camera roll is required!");
      return;
    }

    let pickerResult = await ImagePicker.launchImageLibraryAsync();
    if (pickerResult.cancelled === true) {
      return;
    }
    addImage(item.id, pickerResult.uri);
  };

  return (
    <StyledCard onPress={openImagePickerAsync} style={shadow}>
      <StyledCardMedia>
        <Icon name="upload" color={primaryColor} size={25}></Icon>
      </StyledCardMedia>
      <StyledCardTitle>Edit image</StyledCardTitle>
    </StyledCard>
  );
};

export default TodoDetailsSettings;
