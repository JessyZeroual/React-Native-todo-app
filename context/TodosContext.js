import React from "react";

const TodosContext = React.createContext({});

export const TodosContextProvider = TodosContext.Provider;
export const TodosContextConsumer = TodosContext.Consumer;
export default TodosContext;
