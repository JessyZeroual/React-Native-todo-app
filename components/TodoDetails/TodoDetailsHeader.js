import React, { useState } from "react";
import Icon from "react-native-vector-icons/AntDesign";
import Modal, {
  ModalTitle,
  SlideAnimation,
  ModalContent,
} from "react-native-modals";

import TodoDetailsSettings from "./TodoDetailsSettings";

import {
  ImageWrapper,
  Image,
  Title,
  WrapperHeader,
  Header,
} from "./TodoDetails.styled";

const TodoDetailsHeader = ({ todo }) => {
  const [isVisibleModal, setIsVisibleModal] = useState(false);
  return (
    <>
      <ImageWrapper>
        <Image
          source={{
            uri: "https://reactnative.dev/img/tiny_logo.png",
          }}
        />
      </ImageWrapper>
      <WrapperHeader>
        <Header>
          <Title>{todo.title}</Title>
          <Icon
            style={{ marginLeft: "auto" }}
            name="setting"
            color="grey"
            size={30}
            onPress={() => setIsVisibleModal(true)}
          ></Icon>
          <Modal
            modalTitle={<ModalTitle title={todo.title} />}
            visible={isVisibleModal}
            modalAnimation={
              new SlideAnimation({
                slideFrom: "bottom",
              })
            }
            onTouchOutside={() => {
              setIsVisibleModal(false);
            }}
          >
            <ModalContent>
              <TodoDetailsSettings todo={todo} />
            </ModalContent>
          </Modal>
        </Header>
      </WrapperHeader>
    </>
  );
};

export default TodoDetailsHeader;
