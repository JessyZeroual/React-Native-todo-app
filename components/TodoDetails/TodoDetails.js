import React from "react";
import { ActivityIndicator, View } from "react-native";

import TodoDetailsHeader from "./TodoDetailsHeader";
import TodoDetailsBody from "./TodoDetailsBody";
import { ViewCenter } from "../../style/common.styled";
import { TodoDetailsWrapper } from "./TodoDetails.styled";

const TodoDetails = ({ route }) => {
  const { item } = route.params;

  return (
    <View>
      {item ? (
        <TodoDetailsWrapper>
          <TodoDetailsHeader item={item} />
          <TodoDetailsBody item={item} />
        </TodoDetailsWrapper>
      ) : (
        <ViewCenter>
          <ActivityIndicator size="large" />
        </ViewCenter>
      )}
    </View>
  );
};

export default TodoDetails;
