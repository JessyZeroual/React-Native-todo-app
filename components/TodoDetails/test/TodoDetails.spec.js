import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { ActivityIndicator, View } from "react-native";
import { TodosContextProvider } from "../../../context/TodosContext";
import TodoDetails from "../TodoDetails";

jest.mock("../TodoDetailsBody", () => () => "TodoDetailsHeader");
jest.mock("../TodoDetailsHeader", () => () => "TodoDetailsBody");

const mockaddImage = jest.fn();
const mockUpdateTodo = jest.fn();

const route = {
  params: { item: { id: 1, title: "todo-1" } },
};
const { item } = route.params;

describe("InputCreateItem", () => {
  let renderer;

  beforeEach(() => {
    act(() => {
      renderer = TestRenderer.create(
        <TodosContextProvider
          value={{
            dispatch: { addImage: mockaddImage, _updateTodo: mockUpdateTodo },
          }}
        >
          <TodoDetails route={route} />
        </TodosContextProvider>
      );
    });
  });

  describe("When InputCreateItem is mounted", () => {
    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });

    it("has item in pros with react-navigation", () => {
      expect(renderer.root.props.route.params.item).toEqual(item);
    });
  });
});



