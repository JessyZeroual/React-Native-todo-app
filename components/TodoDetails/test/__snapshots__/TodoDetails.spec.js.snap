// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCreateItem When InputCreateItem is mounted renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomColor": "#ebebef",
          "borderBottomWidth": 1.5,
          "display": "flex",
          "height": "100%",
        },
      ]
    }
  >
    TodoDetailsBody
    TodoDetailsHeader
  </View>
</View>
`;
